crates/backend/src/util.rs

```diff
@@ -0,0 +1,226 @@
+use std::path::Path;
+
+use crate::SupportedBackend;
+use anyhow::Result;
+use dunce::canonicalize;
+use url::Url;
+
+#[derive(PartialEq, Debug)]
+pub struct BackendLocation(String);
+
+impl std::ops::Deref for BackendLocation {
+    type Target = String;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl TryFrom<&str> for BackendLocation {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> std::result::Result<Self, Self::Error> {
+        match (canonicalize(Path::new(value)), Url::parse(value), value) {
```

Comment:
canonicalize returns an error if the path doesn't exist. We can't use this in combination with `init` - here we actually want to create a path, but will run into `Invalid backend location`....